packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/FieldLayoutOptions.tsx

```diff
@@ -165,17 +165,16 @@ const FieldLayoutOptions: FC<Props> = ({ items }) => {
 
                 {yFields.map((field, index) => {
                     const activeField = yActiveField(field);
+                    const yFieldsOptions = activeField
+                        ? [activeField, ...availableYFields]
+                        : availableYFields;
                     return (
                         <AxisFieldDropdown key={`${field}-y-axis`}>
-                            <FieldAutoComplete
-                                fields={
-                                    activeField
-                                        ? [activeField, ...availableYFields]
-                                        : availableYFields
-                                }
-                                activeField={activeField}
-                                onChange={(item) => {
-                                    updateYField(index, getItemId(item));
+                            <FieldSelect
+                                selectedField={activeField}
+                                fieldOptions={yFieldsOptions}
+                                onChange={(newValue) => {
+                                    updateYField(index, newValue ?? '');
```

Comment:
should we be allowing setting to '' or we should do the same as `setXField`

Reply:
I'm not sure it matters functionally, but you're right that they should be consistent. I think this is kinda more 'correct' than setting it to undefined. We aren't controlling these components here (which is a little odd), but for controlled components react doesn't like undefined. So this might be a better default answer than undefined. Although null might be even better. Let me see if Mantine has an opinion.

Reply:
ok sounds good!